src/edu/psu/sweng500/team8/gui/BoardGUI.java

```diff
@@ -252,6 +265,20 @@ public void refreshPencilMarkDisplayOnRelatedCells(Cell cell) {
 			}
 		}
 	}
-	
 
+	/**
+	 * Issue #310
+	 * If a Cell has a user entered number and currently selected, we should not leave it selected
+	 * when we switch to Pencil Mark Mode.
+	 */
+	public void unselectCellWithNumber() {
+		if (null != this.selectedCell) {
+			
+			if (this.selectedCell.getCell().hasNumber()) {

```

Comment:
Suggest combining this

if (this.selectedCell != null && this.selectedCell.getCell().hasNumber()) { //...}